src/Util/ConfigHandler.php

```diff
@@ -40,4 +46,13 @@ public static function setAuthentication(array &$config, array $auth)
     {
         $config[self::CONFIG_NAMESPACE][self::KEY_AUTHENTICATION] = $auth;
     }
+
+    public static function appendConfiguration(array &$config, array $options)
+    {
+        $optionsToAppend = array_filter($options, function($option) {
+            return in_array($option, self::$whitelistedConfiguration, true);
+        }, ARRAY_FILTER_USE_KEY);
```

Comment:
`ARRAY_FILTER_USE_KEY` is supported from PHP 5.6 - we should raise composer.json or avoid using this